### DISCUSSION TITLE
Is there a way to totally disable go2rtc in frigate 12

### DISCUSSION THREAD

**Discussion by tbrausch** [blakeblackshear/frigate#5705] [Ask A Question]
I seem to remember in the early betas that there was a way to disable the embedded version of go2rtc (although I searched and could not find it).  Is this still possible?

**Comment by blakeblackshear** [comment]
It was never possible to disable it. You can choose not to use it if you don't want to use any of the features it provides.

**Reply by kerryland** [reply]
I am running frigate on a 2gb Jetson Nano, and could use all the memory I can get. The ability to disable go2rtc would help.

**Comment by gretel** [comment]
using `go2rtc` without frigate got quite popular hence it would be great to have the option to disable the internal one.

**Reply by NickM-27** [reply]
a number of features in frigate itself won't work if you use go2rtc externally. go2rtc internally also fully supports the same features so there should be no difference.

**Reply by alex-savin** [reply]
But the current version of the frigate running old version of the go2rtc

**Reply by NickM-27** [reply]
> But the current version of the frigate running old version of the go2rtc

you can override it with the current version if you want to, there is no limitation on that.

**Reply by moskovskiy82** [reply]
> > But the current version of the frigate running old version of the go2rtc
> 
> you can override it with the current version if you want to, there is no limitation on that.

How that can be done of using docker compose?

**Reply by blakeblackshear** [reply]
https://docs.frigate.video/configuration/advanced#custom-go2rtc-version

**Reply by kidhasmoxy** [reply]
Is there full support for all the capabilities? go2rtc fails to load as part of Frigate if I use the syntax for applying ffmpeg config to a HomeAssistant camera as seen below. The same config works fine in the go2rtc addon for HA. 
```
streams:
  - "camera.camera_entity": ffmpeg:{input}#video=copy#hardware#audio=opus
```

**Reply by NickM-27** [reply]
that is covered in the docs https://docs.frigate.video/configuration/restream#advanced-restream-configurations

**Comment by erenoglu** [comment]
If I have a separate go2rtc docker and still start frigate docker (using host networking), I'm getting error messages from the built-in go2rtc that address already in use for the go2rtc ports. How can I tell frigate not to start it if it isn't used?

**Reply by NickM-27** [reply]
you can't, given that you can run frigate in host networking I see no reason not to use the built in go2rtc config

**Reply by kerryland** [reply]
Because you are running a 2gb Nano and want to save some memory and cpu cycles...

**Reply by erenoglu** [reply]
No I was just testing some issues on the go2rtc streaming skipping frames on browsers while VLC being flawless on same stream. Anyway I used the custom go2rtc article to install the latest version and testing with it. Thanks for the responses.

**Reply by NickM-27** [reply]
I am suggesting there is no reason to run go2rtc externally when it can be run with all the same features internally. In that case there would be no difference

**Reply by stoneobscurity** [reply]
except the one frigate uses is out of date.

**Reply by NickM-27** [reply]
and you can set frigate to use whatever version you want https://docs.frigate.video/configuration/advanced#custom-go2rtc-version

**Reply by stoneobscurity** [reply]
is that a v13? cuz it did not work for me in v12.

**Reply by NickM-27** [reply]
it works in 0.12, a large amount of users have used this feature as part of troubleshooting and using two way audio

**Reply by stoneobscurity** [reply]
being able to disable it entirely seems like something people want. maybe add it with a disclaimer about "at your own risk".

**Reply by NickM-27** [reply]
add what with a disclaimer? also we have had very few requests to be able to disable it

**Reply by BeeSci** [reply]
I also would prefer the option to disable the built-in go2rtc and use the standard go2rtc addon instead.

**Reply by moskovskiy82** [reply]
Would also love to see this feature
1. Updating the separate go2rtc docker container is a bit easier
2. With all the security implementations it's easier to edit the config and restart it any time

**Reply by anjanms** [reply]
The ability to disable the embedded Go2RTC since it’s unnecessary overhead in my situation would be nice.

Frigate is an addition to my existing NVR setup, for the purpose of detecting objects, triggering automation, and capturing those events. 24/7 recording is handled by FFmpeg, live view (and RTSP restreaming) is handled by Go2RTC.

Having these operate as independent services makes the whole setup more resilient to failures. Restarting or misconfiguring Frigate isn’t going to interrupt 24/7 recording or live view. Likewise Go2RTC failing for one reason or another would only interrupt live view & Frigate.

As an aside; Running the Frigate container as a non-root user is proving difficult, not because of hardware permissions but because of the logging system of all things, though that’s a somewhat related but [different issue](https://github.com/blakeblackshear/frigate/issues/3108#issuecomment-1408207023).

**Reply by Gilrain** [reply]
Ok we can't disable it, but we can eliminate the port conflicts.

Frigate is hard-coded to look for go2rtc on 127.0.0.1. So we move the embedded go2rtc to another localhost (e.g. 127.255.0.1)!

Simply replace the go2rtc part of frigate's config.yml with this:
```
go2rtc:
  # Disable go2rtc by listening on 127.255.0.1
  api:
    listen: "127.255.0.1:1984"
  webrtc:
    listen: "127.255.0.1:8555"
    candidates:
      - 127.255.0.1:8555
  rtsp:
    listen: "127.255.0.1:8554"
```

Frigate will listen to the external go2rtc and blissfully ignore the shipped version (provided docker network is in "host" mode, of course).

**Comment by gretel** [comment]
> How can I tell frigate not to start it if it isn't used?

you can not.

**Comment by gtxaspec** [comment]
+1
Running external go2rtc.

**Reply by ouinouin** [reply]
same feeling, here, the internal version dont seem to support ffmpeg alsa, it can be a good idea to just have a docker environment flag to not start  go2rtc by default. putting a binary in config seems a good idea, but it add some heavy extra steps for just a testing user that miss a functionalty embedded in your version of go2rtc, in the other hand, starting another go2rtc docker is just a copy paste away.

**Comment by MaT-dg** [comment]
+1 vote to be able to disable the built-in go2rtc.

In the latest 2024.11 update of Homeassistant there is also a go2rtc instance built in. So now I have 3 go2rtc's running. I really only need one.
Please reconsider this. Thank you.

**Comment by lucabravi** [comment]
+1 for an option to completely disable go2rtc, I run it outside of the frigate container

**Comment by hugalafutro** [comment]
Still no option to disable go2rtc?

**Comment by hawkeye217** [comment]
If you don't specify a go2rtc section in your frigate config file, go2rtc will not be used for anything in frigate, essentially disabled.  

However, frigate's _internal_ go2rtc is required for enhanced live view (full frame rate and audio), two-way talk, and other streaming features (especially in the upcoming version of frigate). 

Using an external version of go2rtc for these features would make supporting them for users much more difficult. 

Enhanced live view remains optional, of course. 

https://docs.frigate.video/configuration/live#live-view-technologies

**Reply by hugalafutro** [reply]
Thank you for quick answer and explaining it's still used internally. As the others posters I'm now in situation where there were multiple instances of go2rtc (as ha includes it now as well) running and I was trying to reduce system load (it runs on heavily overloaded N100 proxmox shitbox so every bit helps)

**Reply by hawkeye217** [reply]
When unused by frigate, its internal go2rtc just sits completely idle. With no streams defined, I'd imagine its memory footprint is extremely low as well.

**Comment by ideaalab** [comment]
+1 waiting for the option to disable internal go2rtc :)

**Comment by BEFH** [comment]
the internal ffmpeg is completely screwed up, so I am trying to switch to my own docker with ffmpeg and go2rtc, but now people are saying the internal go2rtc is needed to support features. Also, since my switch, the ha addon cannot load. I don't know what to do here.

**Reply by NickM-27** [reply]
you can configure go2rtc to use a custom ffmpeg if you need to, easiest way is to just do that

**Reply by BEFH** [reply]
The problem is that it's hwaccel that's screwed up. I am using a custom
container with a dynamically linked ffmpeg 8, and that just wouldn't work
at all with a static binary.

Here's my go2rtc: https://github.com/BEFH/docker_go2rtc

The big thing is I cannot get anything at all working with the b580 GPU and
the embedded ffmpeg. It consistently crashes with both vaapi and qsv within
the frigate container, but not within my go2rtc container.
